### ISSUE TITLE
[Suggestion] A new category for rapidly flashing things to help people with epilepsy

### ISSUE THREAD

**Issue by bbb651** [ajayyy/SponsorBlock#403]
The title says it all. It's a bit different than the rest of the categories but it can be very helpful for a lot of people.

**Comment by elypter** [comment]
that sounds more than useful enough on its own but it also helps to legitimize the addon and discourages platforms to disallow it.

**Comment by nfd9001** [comment]
This may be extremely useful for anyone who is photosensitive might want to put on a video in a language they can't read well/fully understand (e.g. language learners, people listening to foreign music, ...). Hazards that a viewer *can't* be properly warned about could be lurking around any corner.

**Comment by izzy** [comment]
I just had the same idea after realising this extension could be used for not just this but could be extended to different sort of content that aren't easily or safely consumable by everyone(think other sensory issues like auditory overstim) or could have other adverse reactions. There's a lot of value this extension could have in terms of health and safety.

**Comment by erkinalp** [comment]
Related: #1310 

Could be merged into "intentional sensory overstimulation", with possible "remedies" of *skipping*, *blackout*, *muting*.

**Comment by FlashlightET** [comment]
if they were merged would i be able to set it to skip earrape but not skip flashing lights

**Comment by erkinalp** [comment]
@FlashlightET #1308

**Comment by SethFalco** [comment]
I accidentally made a duplicate issue, but since I got out of my way to write it up anyway, figured I might as well copy some of the info into a comment here. ^-^'

I'd really like to see this as well as it would be a valuable accessibility feature which YouTube has skipped out on. It's important to note that it's not just about epilepsy, but general photosensitivity. One doesn't have to have seizures or the like to find high visual stimulation discomforting.

Some videos that might be covered by a request like this:
* Visual
  * [Ad](https://yewtu.be/watch?v=e-a-_xxYjNM) (Flashes of highly saturated colors.)
  * [How Many Shirts Will Stop a Bullet?](https://yewtu.be/watch?v=F8xIr28fns0) (Flashes at 0:38, see [Reddit comment](https://libreddit.spike.codes/r/Epilepsy/comments/warnn7/youtube_comment_for_flashing_light_warning/).)

I have encountered numerous users who are seeking a solution to this problem which I believe SponsorBlock could solve. Also, just noting some external posts that essentially request this feature from YouTube:
* [Reddit: YT should add a seizure warning section when we upload videos](https://libreddit.spike.codes/r/youtube/comments/m4q4uh/yt_should_add_a_seizure_warning_section_when_we/) (1,300+ upvotes)
* [Reddit: Can they put seizure warnings on any videos that have flashing lights?](https://libreddit.spike.codes/r/roosterteeth/comments/bho0ng/can_they_put_seizure_warnings_on_any_videos_that/) (171 upvotes)

**Comment by PencilVoid** [comment]
Came here to suggest this. I know it's not the main focus of the extension but it would be an incredibly valuable feature. I would suggest using separate categories for photosensitivity warnings and ear rape/screamers.

**Comment by Doksha** [comment]
> I would suggest using separate categories for photosensitivity warnings and ear rape/screamers.

would probably be best to leave as one, if someone isn't bothered by loud noises but is bothered by flashing lights, they can hover over the timeline to see which it is, but the user submitting should be encouraged to black out screen if it is flashing lights, and mute audio if it is loud noises, devs seem to not like adding too many categories anyway, so this is probably best

**Comment by Doksha** [comment]
> Please keep in mind that both of these are for some a mere annoyance but for others a matter of health/safety. It's like an elevator - for most people elevators are a nice feature of the building but for some people they are essential.
> 
> I feel like putting both in the same category and saying "people can figure it out with the timeline" ignores that they are each on their own a distinct overstimulation for certain groups of people and this puts the burden potentially on disabled people. Idk about you but to me that doesn't sound right.

as I said in the next half of the sentence, the main differentiation should be whether the screen is blacked, or the audio is muted.
if the devs won't implement a feature, it doesn't matter what the feature does; putting them separate will make the devs much less likely to implement it, they already have so many double purpose flags, and both should be implemented, I see it as the best option.

**Comment by MissLavender-LQ** [comment]
agreed this would be very helpfully so many videos I watch have flashing in them.of some sort and I just have to look away or manually skip ahead while looking away

I thankfully don't get seizures but it is quite painful

**Comment by levyd75** [comment]
Like others, I just found this thread after coming here to suggest such a category.

Even people _without_ medical conditions might find rapidly flashing lights/colors highly irritating, in line with other issues that SponsorBlock currently covers.  The serious impact on photosensitive individuals only makes the issue more important.

**Comment by ajayyy** [comment]
This may be interesting to people here https://chrome.google.com/webstore/detail/seizafe-epilepsy-alert-fo/goageegpmdbgenkgkhfhnnjmgmgboegi

**Comment by MissLavender-LQ** [comment]
> This may be interesting to people here (it is not by me) https://chrome.google.com/webstore/detail/seizafe-epilepsy-alert-fo/goageegpmdbgenkgkhfhnnjmgmgboegi

fuuck its chrome only

**Comment by MissLavender-LQ** [comment]
might look into this more firefox says it might include payed shit but ill see later
https://addons.mozilla.org/en-US/firefox/addon/epilepsy-blocker/

**Comment by izzy** [comment]
> This may be interesting to people here (it is not by me) https://chrome.google.com/webstore/detail/seizafe-epilepsy-alert-fo/goageegpmdbgenkgkhfhnnjmgmgboegi

Tbh both Chrome and FF extensions seem to both have problems with development. Both haven't seen an update in a questionably long time and while the dev for the Chrome extension has been talking about an "big update" that was way back in January.

**Comment by MisterSheeple** [comment]
I'm really surprised this hasn't been implemented as a category yet. I kinda thought it would be considered common sense. Both this and loudness deserve to be their own categories to serve better accessibility IMO.

**Comment by Wemmy0** [comment]
It's definitely out of the scope of the original premise of the project but because it's become such a popular extension, with most videos you'd want to watch having segments, it's seems only logical to add a loudness and a flashing images category.

**Comment by captainharrie** [comment]
Came here to see if this has been suggested after I just got very suddenly hit with some aggressive flashing that was completely unwarned for until it was already happening and the YTer just said "sorry about the flashing btw" and am now going to be nursing a migraine for the next few days. These days if I watch a video on youtube there seems to be a 50% chance it'll include completely unwarned for flashing, and I'd really like to be able to skip these things automatically like I can with sponsored segments.

I do realise that sponsorblock's original scope was not for this purpose, but the specific purpose of sponsorblock was to skip sponsored segments of the video and we now have categories for skipping jokes, skipping intros and intermissions, and skipping recaps so there is already plenty of precedence for the extension going outside of its original scope and covering the general concept of skipping annoyances in a video, of which flashing/strobing/general eyestrain and loud noises all count as.

So in my opinion, this is within the current scope of the extension, and I'd really appreciate its inclusion.

**Comment by ChrisIsEditing** [comment]
Is this being considered by the team yet? It would be really great if this could be added!

**Comment by Julmik647** [comment]
Epilepsy is in [rejected suggestions](https://wiki.sponsor.ajay.app/w/FAQ/New_categories) (scroll down)

**Comment by xNasuni** [comment]
i refuse to believe there is even more than 1 percent of people who think the extension isn't just community made sections. the statement that having it would be "worse than nothing" because it would give a false sense of security of being completely protected from this harm is completely wrong in my eyes, and is inhumane in my opinion for something that could be so positive. it is definitely better than nothing.

**Comment by Julmik647** [comment]
I have a solutions that it could automaticly detect rapidly flashing things. I think that it can be done even without using AI or machine learning, just some basic set of rules to detect it.

**Comment by 0Andras** [comment]
> Epilepsy is in [rejected suggestions](https://wiki.sponsor.ajay.app/w/FAQ/New_categories) (scroll down)

I don't suffer from epilepsy myself, but what the hell is that reasoning? "False sense of security"? Come the fuck on, not even every sponsor is always properly skipped! This entire thing is built on anything being better than nothing, yet _now_ that's no longer good enough? A simple disclaimer that it shouldn't be treated like it's omniscient would be enough. "It wouldn't be absolutely perfect, so we won't even try" is a colossally shit take.
